plugins/http2_adapter/lib/src/http2_adapter.dart

```diff
@@ -158,6 +152,19 @@ class Http2Adapter implements HttpClientAdapter {
       cancelOnError: true,
     );
 
+    /// Cancel any up/download streams and connections
+    /// if the [CancelToken] is cancelled
+    /// and propagate the [DioException] to the response stream.
+    cancelFuture?.whenComplete(() {
```

Comment:
We have asynchronous gaps between this call and other awaits. Can we do this earlier or separated?